newrelic/structures_newrelic_alert_muting_rule.go

```diff
@@ -16,10 +20,170 @@ func expandMutingRuleCreateInput(d *schema.ResourceData) alerts.MutingRuleCreate
 		createInput.Condition = expandMutingRuleConditionGroup(e.([]interface{})[0].(map[string]interface{}))
 	}
 
-	return createInput
+	if e, ok := d.GetOk("schedule"); ok {
+		schedule, err := expandMutingRuleCreateSchedule(e.([]interface{})[0].(map[string]interface{}))
+		if err != nil {
+			return alerts.MutingRuleCreateInput{}, err
+		}
+		createInput.Schedule = &schedule
+	}
+
+	return createInput, nil
+}
+
+func expandMutingRuleCreateSchedule(cfg map[string]interface{}) (alerts.MutingRuleScheduleCreateInput, error) {
+	schedule := alerts.MutingRuleScheduleCreateInput{}
+
+	if startTime, ok := cfg["start_time"]; ok {
+		rawStartTime := startTime.(string)
+		if rawStartTime != "" {
+			formattedStartTime, err := time.Parse("2006-01-02T15:04:05", rawStartTime)
+			if err != nil {
+				return alerts.MutingRuleScheduleCreateInput{}, err
+			}
+			schedule.StartTime = &alerts.NaiveDateTime{Time: formattedStartTime}
+		}
+	}
+
+	if endTime, ok := cfg["end_time"]; ok {
+		rawEndTime := endTime.(string)
+		if rawEndTime != "" {
+			formattedEndTime, err := time.Parse("2006-01-02T15:04:05", rawEndTime)
+			if err != nil {
+				return alerts.MutingRuleScheduleCreateInput{}, err
+			}
+			schedule.EndTime = &alerts.NaiveDateTime{Time: formattedEndTime}
+		}
+	}
+
+	if timeZone, ok := cfg["time_zone"]; ok {
+		schedule.TimeZone = timeZone.(string)
+	}
+
+	if repeat, ok := cfg["repeat"]; ok {
+		r := repeat.(string)
+		if r != "" {
+			sr := alerts.MutingRuleScheduleRepeat(strings.ToUpper(r))
+			schedule.Repeat = &sr
+
+		}
+	}
+
+	if endRepeat, ok := cfg["end_repeat"]; ok {
+		rawEndRepeat := endRepeat.(string)
+		if rawEndRepeat != "" {
+			formattedEndRepeat, err := time.Parse("2006-01-02T15:04:05", rawEndRepeat)
+			if err != nil {
+				return alerts.MutingRuleScheduleCreateInput{}, err
+			}
+			schedule.EndRepeat = &alerts.NaiveDateTime{Time: formattedEndRepeat}
+		}
+
+	}
+
+	if repeatCount, ok := cfg["repeat_count"]; ok {
+		r := repeatCount.(int)
+		if r > 0 {
```

Comment:
Do we need this `r >0` check?  Don't we already validate that in `ValidateFunc:  validation.IntAtLeast(1)`?

Reply:
This is a good question. I believe this is handling the cases where we don't have repeat set in the config, so converting the nil pointer to an int results in a 0. I want to remove it and see what happens, though. Thanks for pointing this out.